google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -717,7 +718,8 @@ public static final class Builder {
     static final long DEFAULT_COMPRESSION_BYTES_THRESHOLD = 240L;
 
     String topicName;
-    private String endpoint = PublisherStubSettings.getDefaultEndpoint();
```

Comment:
This line was added when endpoint overriding was enabled for the `Publisher`/`Subscriber` [here](https://github.com/googleapis/java-pubsub/commit/5872cbdc7a450737f59fb0037403066855b9ab30#diff-95d902b79e31dfdb688e15f96622f27d560ba70ab1311d9b719ec3cf8b0f79c9R592). If a user sets the endpoint when building the `Publisher`, we propagate that to the `PublisherStubSettings builder`, which is autogenerated. Otherwise, `PublisherStubSettings` uses the `getDefaultEndpoint()` method.

Reply:
To add a bit more context from the GAPICs perspective. Gax-Java now will resolve the endpoint automatically so there is no need to set a default endpoint. One of the parameters that we take into consideration is the user set endpoint which is configured from the Builder's `setEndpoint()`. All these factors (i.e. mtls, user set endpoint, user set universe domain, and more) will be considered when creating the final endpoint to be used.

The reason why we're suggesting to not pass the `getDefaultEndpoint` to Gax is that Gax-Java won't be able to determine if the endpoint passed in is a user configuration or not. The new endpoint logic has user configuration override everything so it will always default to `pubsub.googleapis.com`.